Reqnroll.Generator/Generation/UnitTestFeatureGenerator.cs

```diff
@@ -184,11 +192,108 @@ private void DeclareFeatureInfoMember(TestClassGenerationContext generationConte
+                gherkinDocumentFunc,
+                picklesFunc,
+                new CodePrimitiveExpression(sourceFileLocation));
+
+            cucumberMessagesInitializeMethod.Statements.Add(
+                new CodeMethodReturnStatement(
+                    featureLevelCucumberMessagesExpression));
+
+            try
+            {
+                var messageConverter = new CucumberMessagesConverter(new GuidIdGenerator());
+                var featureSource = Formatters.PayloadProcessing.Cucumber.CucumberMessageTransformer.ToSource(messageConverter.ConvertToCucumberMessagesSource(generationContext.Document));
+                var featureGherkinDocument = messageConverter.ConvertToCucumberMessagesGherkinDocument(generationContext.Document);
+                var featurePickles = messageConverter.ConvertToCucumberMessagesPickles(featureGherkinDocument);
+                var featureGherkinDocumentMessage = Formatters.PayloadProcessing.Cucumber.CucumberMessageTransformer.ToGherkinDocument(featureGherkinDocument);
+                var featurePickleMessages = Formatters.PayloadProcessing.Cucumber.CucumberMessageTransformer.ToPickles(featurePickles);
+
+                // generate a CodeDom expression to create the Source object from the featureSourceMessage
+                var sourceExpression = new CodeObjectCreateExpression(new CodeTypeReference(typeof(Io.Cucumber.Messages.Types.Source), CodeTypeReferenceOptions.GlobalReference),
+                    new CodePrimitiveExpression(featureSource.Uri),
+                    new CodePrimitiveExpression(featureSource.Data),
+                    new CodeFieldReferenceExpression(new CodeTypeReferenceExpression(new CodeTypeReference(typeof(Io.Cucumber.Messages.Types.SourceMediaType), CodeTypeReferenceOptions.GlobalReference)), featureSource.MediaType.ToString()));
+
+                // generate a CodeDom expression to create the GherkinDocument object from the featureGherkinDocumentMessage
+                var gherkinDocumentExpressionGenerator = new CucumberGherkinDocumentExpressionGenerator();
+                var gherkinDocumentExpression = gherkinDocumentExpressionGenerator.GenerateGherkinDocumentExpression(featureGherkinDocumentMessage);
+
+                // generate a CodeDom expression to create the Pickles object from the featurePickleMessages
+                var pickleExpressionGenerator = new CucumberPicklesExpressionGenerator();
+                var picklesExpression = pickleExpressionGenerator.GeneratePicklesExpression(featurePickleMessages);
+
+                // wrap these expressions in Func<T>
+
+                sourceFactoryMethod.Statements.Add(new CodeMethodReturnStatement(sourceExpression));
+
+                gherkinDocumentFactoryMethod.Statements.Add(new CodeMethodReturnStatement(gherkinDocumentExpression));
+
+                picklesFactoryMethod.Statements.Add(new CodeMethodReturnStatement(picklesExpression));
+            }
+            catch (Exception e)
```

Comment:
(4) This Try-Catch is not effective, because if there is an exception, the generatred methods will have no return statement and therefore will not even compile. (Try adding a throw statement to the beginning of the try block to repro.) I think we need to have a prepared variable with a "return null" statement outside of the try and override that variable in the try block and after the catch just add the variable to the method statement. And of course the same for all 3 methods.